CMakeLists.txt

```diff
@@ -33,90 +11,98 @@ INCLUDE_DIRECTORIES(
         /usr/include
 )
 
-if(CMAKE_BUILD_TYPE STREQUAL "")
-    set(CMAKE_BUILD_TYPE Release)
-endif()
-
-set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -Wall -std=${CMAKE_CXX_STD_FLAGS}")
-set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -Wall -std=${CMAKE_CXX_STD_FLAGS}")
-
-
 # lib directories
 LINK_DIRECTORIES(
         ${PROJECT_BINARY_DIR}/
         /usr/local/lib
         /usr/lib
 )
 
-AUX_SOURCE_DIRECTORY(trantor/utils DIR_SRC)
-AUX_SOURCE_DIRECTORY(trantor/net DIR_SRC)
-AUX_SOURCE_DIRECTORY(trantor/net/inner DIR_SRC)
-find_package (OpenSSL)
-if(OpenSSL_FOUND)
-    #add_definitions(-DUSE_OPENSSL)
-    set(DEFS "USE_OPENSSL;${DEFS}")
+SET(public_net_headers
+    trantor/net/EventLoop.h
+    trantor/net/EventLoopThread.h
+    trantor/net/EventLoopThreadPool.h
+    trantor/net/InetAddress.h
+    trantor/net/TcpClient.h
+    trantor/net/TcpConnection.h
+    trantor/net/TcpServer.h
+    trantor/net/callbacks.h)
+    
+SET(public_utils_headers
+    trantor/utils/AsyncFileLogger.h
+    trantor/utils/ConcurrentTaskQueue.h
+    trantor/utils/Date.h
+    trantor/utils/Funcs.h
+    trantor/utils/LockFreeQueue.h
+    trantor/utils/LogStream.h
+    trantor/utils/Logger.h
+    trantor/utils/MsgBuffer.h
+    trantor/utils/NonCopyable.h
+    trantor/utils/ObjectPool.h
+    trantor/utils/SerialTaskQueue.h
+    trantor/utils/TaskQueue.h
+    trantor/utils/TimingWheel.h)
+
+SOURCE_GROUP( "Public API" FILES ${public_net_headers} ${public_utils_headers})
+
+SET(trantor_sources
+    trantor/utils/AsyncFileLogger.cc
+    trantor/utils/ConcurrentTaskQueue.cc
+    trantor/utils/Date.cc
+    trantor/utils/LogStream.cc
+    trantor/utils/Logger.cc
+    trantor/utils/MsgBuffer.cc
+    trantor/utils/SerialTaskQueue.cc
+    trantor/utils/TimingWheel.cc
+    trantor/net/EventLoop.cc
+    trantor/net/EventLoopThread.cc
+    trantor/net/EventLoopThreadPool.cc
+    trantor/net/InetAddress.cc
+    trantor/net/TcpClient.cc
+    trantor/net/TcpServer.cc
+    trantor/net/inner/Acceptor.cc
+    trantor/net/inner/Channel.cc
+    trantor/net/inner/Connector.cc
+    trantor/net/inner/Poller.cc
+    trantor/net/inner/Socket.cc
+    trantor/net/inner/TcpConnectionImpl.cc
+    trantor/net/inner/Timer.cc
+    trantor/net/inner/TimerQueue.cc)
+
+FIND_PACKAGE(OpenSSL)
+IF(OpenSSL_FOUND)
     include_directories(${OPENSSL_INCLUDE_DIR})
-    link_libraries(${OPENSSL_LIBRARIES})
-    AUX_SOURCE_DIRECTORY(trantor/net/ssl DIR_SRC)
-endif()
+    LINK_LIBRARIES(${OPENSSL_LIBRARIES})
+    SET(trantor_sources ${trantor_sources} trantor/net/ssl/SSLConnection.cc)
+ELSE()
+    SET(trantor_sources ${trantor_sources} trantor/net/ssl/SSLConnectionSkipped.cc)
+ENDIF()
 
 IF (CMAKE_SYSTEM_NAME MATCHES "Linux")
```

Comment:
I'm wondering if this would be better to do in-source, to include the right file depending on the OS, instead of having cmake handle it.  As it stands you won't be able to cross-compile to linux from something else, if that is a consideration you want to worry about...

Reply:
Yes, you are right, I will find a better way. Thanks!